pytest_html/plugin.py

```diff
@@ -198,6 +198,8 @@ def append_extra_html(self, extra, extra_index, test_index):
 
             elif extra.get('format') == extras.FORMAT_TEXT:
                 content = extra.get('content')
+                if isinstance(content, bytes):
+                    content = content.decode("utf-8")
```

Comment:
In current implementation if content of extra text contains bytes string with utf-8 encoded chars it will raise UnicodeDecodeError. So this fix will always decode bytes strings for extra text.
For example: `"\xe2\x80\xb1" -> u"\2031"`